server/src/main/java/io/spine/server/event/MessageEnrichment.java

```diff
@@ -140,11 +139,18 @@ private void setFields(Message.Builder builder, S sourceMessage, C context) {
                     fieldFunctions.get(sourceFieldClass);
             Collection<FieldDescriptor> targetFields = fieldMap.get(srcField);
             for (FieldDescriptor targetField : targetFields) {
-                Optional<EnrichmentFunction<?, ?, ?>> function =
-                        firstThat(functions,
-                                  SupportsFieldConversion.of(sourceFieldClass,
-                                                             Field.getFieldClass(targetField)));
-                EnrichmentFunction fieldEnrichment = function.get();
+                SupportsFieldConversion conversion =
+                        SupportsFieldConversion.of(sourceFieldClass,
+                                                   Field.getFieldClass(targetField)
+                        );
+                Optional<EnrichmentFunction<?, ?, ?>> function = firstThat(functions, conversion);
+                EnrichmentFunction fieldEnrichment = function
+                        .orElseThrow(() -> newIllegalStateException(
+                                "Unable to get enrichment for the conversion from message field " +
+                                        "of type `%s` to enrichment field of type `%s`.",
+                                conversion.messageFieldClass(),
+                                conversion.enrichmentFieldClass()
+                        ));
```

Comment:
Please split this method into pieces. It's way too long.

Reply:
It is. The change was in-place solution for direct getting of `Optional` value. This method also contains nested loops. Refactoring it would probably require a Method Object. 

Together with other enrichment-related classes placed at the root of the `event` package, I suggest to perform this refactoring as a separate PR. Here's [the issue](https://github.com/SpineEventEngine/core-java/issues/945).